src/libraries/Common/src/Interop/OSX/Swift.Runtime/Foundation.cs

```diff
@@ -0,0 +1,217 @@
+
+        internal unsafe Data(UnsafeRawPointer pointer, nint count)
+        {
+            this = Foundation.PInvoke_Data_InitWithBytes(pointer, count);
+        }
+
+        internal readonly nint Count => Foundation.PInvoke_Data_GetCount(this);
+
+        internal unsafe void CopyBytes(UnsafeMutablePointer<byte> buffer, nint count)
+        {
+            Foundation.PInvoke_Data_CopyBytes(buffer, count, this);
+        }
+
+        public void* Metadata() => metadata;
+    }
+
+    /// <summary>
+    /// Represents Swift Foundation.DataProtocol in C#.
+    /// </summary>
+    internal unsafe partial struct DataProtocol : ISwiftProtocol
+    {
+        public void* WitnessTable(ISwiftType type)
+        {
+            return Foundation.GetProtocolWitnessTable(type.Metadata(), "$s10Foundation4DataVAA0B8ProtocolAAMc");
+        }
+    }
+
+    /// <summary>
+    /// Represents Swift Foundation.ContiguousBytes in C#.
+    /// </summary>
+    internal unsafe partial struct ContiguousBytes : ISwiftProtocol
+    {
+        public void* WitnessTable(ISwiftType type)
+        {
+            return Foundation.GetProtocolWitnessTable(type.Metadata(), "$s10Foundation4DataVAA15ContiguousBytesAAMc");
+        }
+    }
+
+    /// <summary>
+    /// Swift Foundation PInvoke methods in C#.
+    /// </summary>
+    internal static partial class Foundation
+    {
+        internal const string Path = "/System/Library/Frameworks/Foundation.framework/Foundation";
+
+        [LibraryImport(Path, EntryPoint = "$s10Foundation4DataV5bytes5countACSV_SitcfC")]
+        [UnmanagedCallConv(CallConvs = [ typeof(CallConvSwift) ])]
+        internal static unsafe partial Data PInvoke_Data_InitWithBytes(UnsafeRawPointer pointer, nint count);
+
+        [LibraryImport(Path, EntryPoint = "$s10Foundation4DataV5countSivg")]
+        [UnmanagedCallConv(CallConvs = [ typeof(CallConvSwift) ])]
+        internal static unsafe partial nint PInvoke_Data_GetCount(Data data);
+
+        [LibraryImport(Path, EntryPoint = "$s10Foundation4DataV9copyBytes2to5countySpys5UInt8VG_SitF")]
+        [UnmanagedCallConv(CallConvs = [ typeof(CallConvSwift) ])]
+        internal static unsafe partial void PInvoke_Data_CopyBytes(UnsafeMutablePointer<byte> buffer, nint count, Data data);
+
+        [LibraryImport(Path, EntryPoint = "swift_getWitnessTable")]
+        [UnmanagedCallConv(CallConvs = [ typeof(CallConvSwift) ])]
+        internal static unsafe partial void* PInvoke_Swift_GetWitnessTable(void* conformanceDescriptor, void* typeMetadata, void* instantiationArgs);
+
+        [LibraryImport(Path, EntryPoint = "$s10Foundation4DataVMa")]
+        [UnmanagedCallConv(CallConvs = [ typeof(CallConvSwift) ])]
+        internal static unsafe partial void* PInvoke_Data_GetMetadata();
+
+        internal static unsafe void* GetProtocolWitnessTable(void* metadata, string conformanceDescriptorSymbol)
+        {
+            IntPtr handle = DynamicLibraryLoader.dlopen(Path, DynamicLibraryLoader.RTLD_NOW);
```

Comment:
Check for errors - handle can come back IntPtr.Zero and can conformance descriptor.